cgo/cgo_test.go

```diff
@@ -49,6 +111,8 @@ func TestCGo(t *testing.T) {
 				t.Fatal("could not parse Go source file:", err)
 			}
 
+			checkBuildConstraint(t, f)
```

Comment:
Well, honestly, I think it would be easier to understand if you just checked for that file specifically. For example (untested):

```go
minorVersion, err := strconv.Atoi(strings.Split(runtime.Version, ".")[1])
if err != nil {
    t.Fatal("could not parse Go version", runtime.Version)
}
// ...
if name == "errors" && minorVersion < 16 {
    t.Skip()
}
```

That's a very simple test and tests exactly what is needed. In fact, checking for GOOS/GOARCH should not be necessary in CGo tests because we use a fixed GOOS/GOARCH. The only things that we _don't_ control are the libclang version (which shouldn't affect the result much) and the Go version (which can certainly affect the results, for example due to formatting changes, see `normalizeResult`).

Reply:
I don't like parsing `runtime.Version()` as the value is not super consistent if you have a dev version, or even alpha/beta/rc. It's safer to check the build context, and I threw the other ones in because they were there and easy to do, but if you think that's too much, I can roll that back.

Reply:
> the value is not super consistent if you have a dev version, or even alpha/beta/rc

Good point.

Regarding the other values, I think @aykevl is just trying to keep it as simple as possible for the moment, and avoid adding anything we're not using right now. Probably removing at present is a good idea, if you do not mind.

Reply:
> the value is not super consistent if you have a dev version, or even alpha/beta/rc

Agreed, however this is only for tests (which we normally only run on release versions). So I don't think it is that much of an issue.

That said, `build.Default.ReleaseTags` is interesting. Perhaps you could simply check whether `go1.16` is in the list? That should be a very simple test and avoids the issue with dev versions.

Reply:
> I think @aykevl is just trying to keep it as simple as possible for the moment, and avoid adding anything we're not using right now.

Yes, exactly. Usually the simpler something is, the less likely it is to break and the easier it is to maintain.